lib/ddq-backend-mock.js

```diff
@@ -0,0 +1,304 @@
+        record.recordId.requeued = true;
+    }
+
+
+    /**
+     *
+     */
+    class DdqBackendMock extends EventEmitter {
+        /**
+         * @param {Object} backendConfig
+         */
+        constructor(backendConfig) {
+            super();
+            this.config = backendConfig;
+            this.instanceId = null;
+            this.storedData = {};
+            this.poller = null;
+        }
+
+
+        /**
+         * @param {Function} callback
+         */
+        close(callback) {
+            if (this.config.closeFail) {
+                callback(new Error("Could not close connection."));
+            } else {
+                callback();
+            }
+        }
+
+
+        /**
+         *
+         */
+        connect() {
+            console.log("Connecting to database...");
+        }
+
+
+        /**
+         * @param {Object} record
+         */
+        deleteData(record) {
+            delete this.storedData[record.id];
+        }
+
+
+        /**
+         * Want to make an instance id and return if we need to.
+         *
+         * @param {Function} callback
+         */
+        getId(callback) {
+            if (this.instanceId) {
+                callback(this.instanceId);
+            } else {
+                crypto.randomBytes(16, (err, id) => {
+                    if (err) {
+                        callback();
+                    }
+                    this.instanceId = id.toString("hex");
+                    callback(id);
+                });
+            }
+        }
+
+
+        /**
+         * Gets a particular record from stored data.
+         *
+         * @param {string} recordId
+         * @return {(Ddq~record|null)} not finding a record returns nothing
+         */
+        getRecord(recordId) {
+            if (this.storedData[recordId]) {
+                return this.storedData[recordId];
+            }
+
+            return null;
+        }
+
+
+        /**
+         * Looks to see if there are any messages in the storeData and creates
+         * the wrapped message.
+         *
+         * @return {(Ddq~wrappedMessage|null)}
+         */
+        getWrappedMessage() {
+            var file, files, keys;
+
+            files = this.readData();
+            keys = Object.keys(files);
+
+            if (keys.length) {
+                // eslint-disable-next-line no-bitwise
+                file = files[keys[keys.length * Math.random() << 0]];
+
+                if (file && !file.isProcessing) {
+                    files[file.id].isProcessing = true;
+                    this.writeRecord(file);
+
+                    return {
+                        heartbeat: heartbeat.bind(null, this, file.id),
+                        heartbeatKill: heartbeatKill.bind(null, this.file.id),
+                        message: file.message,
+                        remove: remove.bind(null, this, file.id),
+                        requeue: requeue.bind(null, this, file.id)
```

Comment:
On remove or requeue, make sure to do another immediate poll.  Only one needs to be sent at that time, not all of them.